lib/network/utils.ts

```diff
@@ -32,20 +31,37 @@ export function arsenalErrorKMIP(err: string | Error) {
 
 const allowedKmsErrorCodes = Object.keys(allowedKmsErrors) as unknown as (keyof typeof allowedKmsErrors)[];
 
+// Local AWSError type for compatibility with v3 error handling
+export type AWSError = Error & {
```

Comment:
I'm pretty sure you can remove this and use the `S3ServiceException` type, this is exposed by the `@aws-sdk/client-s3` module and all exceptions related to S3 are extending this base type, it has all the fields you have here, with the advantage of being typed & documented